src/server/scripts/Spells/spell_item.cpp

```diff
@@ -258,43 +258,87 @@ class spell_item_goblin_gumbo_kettle : public SpellScriptLoader
     }
 };
 
-enum mithrilSpurs
+enum MountSpells
 {
-    SPELL_MITHRIL_SPEED         = 59916,
+    SPELL_CARROT_ON_A_STICK_EFFECT = 48402,
+    SPELL_RIDING_CROP_EFFECT = 48383,
+    SPELL_MITHRIL_SPURS_EFFECT = 59916,
+    SPELL_MITHRIL_SPURS = 7215,
+    SPELL_MOUNT_SPEED_CARROT = 48777,
+    SPELL_MOUNT_SPEED_RIDING = 48776
 };
 
-class spell_item_mithril_spurs : public SpellScriptLoader
+// 48777 - Mount Speed (11122: Carrot on a Stick)
+// 48776 - Mount Speed (25653: Riding Crop / 32863: Skybreaker Whip - "Property of the Top Orc")
+// 7215 - Mithril Spurs (7969 Mithril Spurs)
+class spell_item_with_mount_speed : public SpellScriptLoader
 {
 public:
-    spell_item_mithril_spurs() : SpellScriptLoader("spell_item_mithril_spurs") { }
+    spell_item_with_mount_speed() : SpellScriptLoader("spell_item_with_mount_speed") { }
 
-    class spell_item_mithril_spurs_AuraScript : public AuraScript
+    class spell_item_with_mount_speed_AuraScript : public AuraScript
     {
-        PrepareAuraScript(spell_item_mithril_spurs_AuraScript);
+        PrepareAuraScript(spell_item_with_mount_speed_AuraScript);
 
-        void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        void OnApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
         {
             Unit* target = GetTarget();
             if (target->getLevel() <= 70)
-                target->AddAura(SPELL_MITHRIL_SPEED, target);
+            {
+                uint32 spellId = 0;
+
+                switch (m_scriptSpellId)
+                {
+                case SPELL_MOUNT_SPEED_CARROT:
+                    spellId = SPELL_CARROT_ON_A_STICK_EFFECT;
+                    break;
+                case SPELL_MITHRIL_SPURS:
+                    spellId = SPELL_MITHRIL_SPURS_EFFECT;
+                    break;
+                case SPELL_MOUNT_SPEED_RIDING:
+                    spellId = SPELL_RIDING_CROP_EFFECT;
+                    break;
+                default:
+                    return;
+                }
+
+                target->CastSpell(target, spellId, aurEff);
+            }
         }
 
         void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
         {
             Unit* target = GetTarget();
-            target->RemoveAurasDueToSpell(SPELL_MITHRIL_SPEED);
+            uint32 spellId = 0;
+
+            switch (m_scriptSpellId)
+            {
+            case SPELL_MOUNT_SPEED_CARROT:
+                spellId = SPELL_CARROT_ON_A_STICK_EFFECT;
+                break;
+            case SPELL_MITHRIL_SPURS:
+                spellId = SPELL_MITHRIL_SPURS_EFFECT;
+                break;
+            case SPELL_MOUNT_SPEED_RIDING:
+                spellId = SPELL_RIDING_CROP_EFFECT;
+                break;
+            default:
+                return;
+            }
```

Comment:
this part is duplicate, so why not wrapping this into a method instead?

Reply:
Cuz I dont know how to do that

Reply:
I can help you later, please remind me

Reply:
@Kitzunu see if the following works for you: https://github.com/Kitzunu/azerothcore-wotlk/pull/2

Reply:
@FrancescoBorzi this does not work

Reply:
Yep this works =) Removed the "this" as it was not neccessary